csharp/src/Ice/TraceUtil.cs

```diff
@@ -2,316 +2,156 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-using ZeroC.Ice;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Globalization;
+using System.IO;
 
-namespace IceInternal
+namespace ZeroC.Ice
 {
     internal static class TraceUtil
     {
-        internal static void TraceSend(Communicator communicator, IList<System.ArraySegment<byte>> buffer)
+        internal static void TraceSendRequest(Communicator communicator, OutgoingRequestFrame request, int size,
+            int requestId, byte compress) =>
+            TraceRequest("sending request", communicator, size, requestId, compress, request.Identity,
+                request.Facet,
+                request.Operation,
+                request.IsIdempotent ? OperationMode.Idempotent : OperationMode.Normal,
+                request.Context,
+                request.Encoding);
+
+        internal static void TraceReceivedRequest(Communicator communicator,
+            IncomingRequestFrame request, int size, int requestId, byte compress) =>
+            TraceRequest("received request", communicator, size, requestId, compress, request.Identity,
+                request.Facet,
+                request.Operation,
+                request.IsIdempotent ? OperationMode.Idempotent : OperationMode.Normal,
+                request.Context, request.Encoding);
```

Comment:
Shouldn't the last parameter be on a new line?

Reply:
Fixed